### PR TITLE
Adding check function to check relation data has proper broker_rsp

### PR DESCRIPTION
**Pull request by xtrusia**
Please find below LP description if it works for this PR

https://bugs.launchpad.net/charm-nova-compute/+bug/1835045

### CODE REVIEW

File: charmhelpers/contrib/storage/linux/ceph.py [hunk 1]
```diff
@@ -1482,13 +1482,21 @@ def send_request_if_needed(request, relation='ceph'):
             relation_set(relation_id=rid, broker_req=request.request)
 
 
+def has_broker_rsp(rid=None, unit=None):
+    rdata = relation_get(rid=rid, unit=unit) or {}
```
Comment (ajkavanagh) [inline charmhelpers/contrib/storage/linux/ceph.py]: This function needs a docstring please, something along the lines of:

```python
"""Return True if the broker_rsp key is 'truthy' (i.e. set to semeting) in the relation data.

:param rid: The relation to check (default of None means current relation)
:type rid: Union[str, None]
:param unit: The remote unit to check (default of None means current unit)
:type unit: Union[str, None]
:returns: True if broker key exists and is set to something 'truthy'
:rtype: bool
"""
```

File: charmhelpers/contrib/storage/linux/ceph.py [hunk 2]
```diff
@@ -1482,13 +1482,21 @@ def send_request_if_needed(request, relation='ceph'):
...
+    broker_rsp = rdata.get(get_broker_rsp_key())
+    if not broker_rsp:
```
Comment (ajkavanagh) [inline charmhelpers/contrib/storage/linux/ceph.py]: It's generally nicer to work with positives:  i.e.

```python
    return True if broker_rsp else False
```

File: charmhelpers/contrib/storage/linux/ceph.py (hunk 1 above)
Comment (xtrusia) [reply charmhelpers/contrib/storage/linux/ceph.py]: Thanks

File: charmhelpers/contrib/storage/linux/ceph.py (hunk 2 above)
Comment (xtrusia) [reply charmhelpers/contrib/storage/linux/ceph.py]: Thanks!

**Comment by xtrusia** [comment]
How can I remove the other 5 commits, some of them are already merged to the master

**Comment by ajkavanagh** [comment]
@xtrusia so, starting from your branch:

```bash
git fetch --all
git rebase <charm-helpers-repo>/master
git push <my-remote> -f
```

This will rebase your branch

**Comment by xtrusia** [comment]
> @xtrusia so, starting from your branch:
> 
> ```shell
> git fetch --all
> git rebase <charm-helpers-repo>/master
> git push <my-remote> -f
> ```
> 
> This will rebase your branch

Thanks a lot

**Comment by xtrusia** [comment]
This is merged in the other PR

**Pull request closed without merging**